plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCreateTableFailureMetadataNotCleanedUp.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.glue;
+
+import org.testng.annotations.Test;
+
+@Test(singleThreaded = true)
+public class TestIcebergGlueCreateTableFailureMetadataNotCleanedUp
+        extends BaseIcebergGlueCreateTableFailureTest
+{
+    private static final String EXCEPTION_MESSAGE = "Test-simulated metastore runtime exception";
+
+    @Override
+    protected void throwCreateTableException()
+    {
+        throw new RuntimeException(EXCEPTION_MESSAGE);
```

Comment:
I think an `OperationTimeoutException` is a valid situation where we may need to not cleanup the metadata. Note that this is an `AmazonServiceException` - this may need adjustment of the logic in `GlueIcebergTableOperations`.